.pre-commit-config.yaml

```diff
@@ -0,0 +1,100 @@
+repos:
+        - repo: https://github.com/pre-commit/pre-commit-hooks
+          rev: v4.0.1
+          hooks:
+                - id: check-yaml
+                - id: end-of-file-fixer
+                  exclude: \.txt$
+                - id: trailing-whitespace  # Remove trailing whitespaces
+                - id: check-merge-conflict
+        # Fix common spelling mistakes
+        - repo: https://github.com/codespell-project/codespell
+          rev: v2.0.0
```

Comment:
All those version numbers will possibly have to be updated regularly?

Reply:
Yes, I don't really know why the versioning is fixed. I copied the pandas config and made sure all are updated.

Reply:
I just had a look and it seems to be a decision in order to make sure the linting is consistent (if upstream is changed, it shouldn't change the lint unless specified)